src/util/file.c

```diff
@@ -40,6 +40,29 @@
 #include "lib/message.h"
 #include "lib/privilege.h"
 
+int has_perm(int level, struct stat file_stat) {
```

Comment:
@bbockelm So I actually wrote this function just really quickly as an example for what it *could* look like, however I never tested it at all. The idea of the function would be that it checks whether or not a user has the permission specified by level (4 = read; 2 = write; 1 = execute). Given one of these numbers as a level, it would then compare it to the owner field, the group field, and the other field from the files permission bitfield. I wasn't sure if the comparison using bitwise and would automatically convert to octal or not, but this was never meant to necessarily be merged as is, rather as me just throwing out some example code.